dashboard/app/models/pd/payment_term.rb

```diff
@@ -66,11 +68,72 @@ def self.for_workshop(workshop)
     found_payment_terms[0]
   end
 
+  def date_range
+    start_date..(end_date || Date::Infinity.new)
+  end
+
   private
 
   def sufficient_contract_terms
     unless per_attendee_payment? || fixed_payment
       errors.add(:base, 'Must have either per attendee payment or fixed payment')
     end
   end
+
+  def truncate_previous_term
+    extant_payment_terms = Pd::PaymentTerm.where(
+      regional_partner: regional_partner,
+      course: course,
+      subject: subject
+    )
+
+    # Payment terms that conflict with the start date
+    new_term_range = start_date..(end_date || Date::Infinity.new)
+    extant_payment_terms.each do |old_payment_term|
+      old_term_range = old_payment_term.start_date..(old_payment_term.end_date || Date::Infinity.new)
+
+      # Four possible overlaps, illustrated by this ascii art
+      # Case 1
+      #  <---Old--->
+      #       <---New--->
+
+      # Case 2
+      #  <---Old------------>
+      #       <---New--->
+
+      # Case 3
+      #       <---Old--->
+      #  <---New--->
+
+      # Case 4
+      #      <---Old--->
+      #  <----------New--->
+      next unless new_term_range.overlaps?(old_term_range)
+      if old_payment_term.start_date < start_date
+        if end_date.nil? || (old_payment_term.end_date && old_payment_term.end_date < end_date)
+          # Case 1
+          old_payment_term.update(end_date: start_date)
+        else
+          # Case 2
+          Pd::PaymentTerm.create!(
+            start_date: end_date,
+            end_date: old_payment_term.end_date,
+            regional_partner: regional_partner,
+            course: course,
+            subject: subject,
+            properties: old_payment_term.properties
+          )
+          old_payment_term.update(end_date: start_date)
```

Comment:
This still doesn't seem right. The way it stands, this will update the end date of a term that starts and ends before mine (no overlap). 

E.g. 
Old: June 1 - July 1
New: Aug 1 - Sept 1

Resulting Old: June 1 - Aug 1